R/cube-transforms.R

```diff
@@ -86,37 +110,187 @@ setMethod("subtotalArray", "CrunchCube", function(x, headings = FALSE) {
 #'
 #' @aliases subtotalArray
 #' @export
-applyTransforms <- function (x, array = cubeToArray(x), ...) {
-    # if any of the dimensions are subvariables, don't even attempt to calculate
-    # transforms or insertions.
-    if (any(getDimTypes(x) == "ca_items")) {
-        return(array)
-    }
+applyTransforms <- function (x,
+                             array = cubeToArray(x),
+                             transforms_list = transforms(x),
+                             dims_list = dimensions(x),
+                             useNA = x@useNA,
+                             ...) {
+    trans_indices <- which(vapply(transforms_list, Negate(is.null), logical(1)))
+    
+    # Try to calculate the transforms for any dimensiton that have them, but
+    # fail silently if they aren't calcuable. We use a for loop so that failing
+    # one dimension doesn't break others. We could possibly use something like
+    # abind::abind here and bind together the insertion vectors in array form,
```

Comment:
I think this is a good use of a for loop. Another dependency which would let us handle this situation a bit more cleanly is [purrr::safely](https://purrr.tidyverse.org/reference/safely.html). I think that purrr would help in a lot of other areas as well, so it might be a better dependency than `abind`. I guess the other way you could do with it is to use an `apply` function, and then return something which could be filtered out of the returned list, but I'm not sure that's any clearer than the loop.